### PR TITLE
refactor: Rewrite updateFilteredPolicies method

### PR DESCRIPTION
**Pull request by basakest**

### CODE REVIEW

**Comment by hsluoyz** [comment]
@techoner plz review

File: src/Adapter.php [hunk 1]
```diff
@@ -91,6 +91,44 @@ public function removePolicy($sec, $ptype, $rule): void
...
+            $removedRules = [];
+            $rows = $entity->all();
+            foreach ($rows as $row) {
+                unset($row->id);
+                unset($row->ptype);
+                $row = $row->toArray();
+                $row = array_filter($row, function ($value) {
+                    return !is_null($value) && $value !== '';
+                });
+                $removedRules[] = array_values($row);
```
Comment (leeqvip) [inline src/Adapter.php]: @basakest For filtering rules, please refer to: https://github.com/php-casbin/dbal-adapter/blob/master/src/Adapter.php#L435

File: src/Adapter.php [hunk 2]
```diff
@@ -91,6 +91,44 @@ public function removePolicy($sec, $ptype, $rule): void
...
+            }
+            $this->removePolicies($sec, $ptype, $removedRules);
```
Comment (leeqvip) [inline src/Adapter.php]: Here again use `removePolicies` to delete one by one, which will affect performance .

**Review by leeqvip — CHANGES_REQUESTED** [review]

File: src/Adapter.php [hunk 3]
```diff
@@ -91,6 +91,42 @@ public function removePolicy($sec, $ptype, $rule): void
...
+            }
+            $this->table->deleteAll($conditions);
+        });
+
+        return $removedRules;
+    }
+
     public function removeFilteredPolicy($sec, $ptype, $fieldIndex, ...$fieldValues) :void
     {
         throw new CasbinException('not implemented');
```
Comment (leeqvip) [inline src/Adapter.php]: @basakest Please implements it use `_removeFilteredPolicy()`.

**Review by leeqvip — CHANGES_REQUESTED** [review]

**Review by leeqvip — APPROVED** [review]

**Pull request merged**